pkg/apiserver/handlers.go

```diff
@@ -331,6 +331,9 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	}
 
+	// For types that aren't root scoped, we default to the "default" namespace if none is specified for backwards compatibility
+	namespaceUnlessRootScoped := ""
```

Comment:
When accessing:

/namespaces/foo

I thought the namespace context had to be `foo` to allow for proper policy evaluation.

This appears to change that.  @deads2k - did we not run into this issue, and did we resolve it elsewhere?